### PR TITLE
Add hcesar.map

### PR DESCRIPTION
**Pull request by mareuo**
This adds hcesar layout, for portuguese speaking countries.

### CODE REVIEW

**Comment by legionus** [comment]
Please explain what kind of hcesar keyboard layout is this?
Do you have links with description or images?

Also kbd uses DCO to format patches. Please add `Signed-off-by` tag in the commit message. Everything else looks good.

https://github.com/legionus/kbd/blob/master/docs/process/howto-contribute.md#sending-patches

**Comment by mareuo** [comment]
Hcesar is a non-qwerty keyboard optimized for Portuguese, much like the Turkish F for Turkish. 
It was made until 1985, when last typewriter factory closed.

It's available here: https://help.keyman.com/keyboard/hcesar/1.2/hcesar
The layout looks like this: https://en.wikipedia.org/wiki/HCESAR#/media/File:KB-Hcesar.svg

File: data/keymaps/i386/hcesar/hcesar.map [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# based on keyman Hcesar (Portugal) layout
+# Derived from pt-latin1.map 
+# With fixes for keycode 43 from pmmm@camoes.rnl.ist.utl.pt, 980626.
+# With fixes for keycodes 27 and 43 from Francisco Miguel Pedroso Honório Colaço
+# (fcolaco@wolf.fis.uc.pt).
+#
+charset "iso-8859-1"
+keymaps 0-2,4,6,8,12
+alt_is_meta
+include "qwerty-layout"
```
Comment (dmage) [inline data/keymaps/i386/hcesar/hcesar.map:11]: Do you need to include qwerty-layout if it's a non-qwerty layout?

File: data/keymaps/i386/hcesar/hcesar.map [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	alt	keycode  40 = Meta_apostrophe
+keycode  41 = backslash        bar
+	control	keycode  41 = nul
+	alt	keycode  41 = Meta_grave
+keycode  42 = Shift
+keycode  43 = dead_tilde       dead_circumflex   asciitilde  asciicircum
+	control keycode  43 = Control_backslash
+	alt     keycode  43 = Meta_backslash
+keycode  44 = y
+keycode  45 = +ccedilla         Ccedilla
```
Comment (dmage) [inline data/keymaps/i386/hcesar/hcesar.map:84]: Perhaps it should be `+ccedilla +Ccedilla`?

File: data/keymaps/i386/hcesar/hcesar.map [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+keycode  39 = x     
+	alt	keycode  39 = Meta_semicolon
+keycode  40 =   comma            semicolon
+	control	keycode  40 = Control_g
+	alt	keycode  40 = Meta_apostrophe
+keycode  41 = backslash        bar
+	control	keycode  41 = nul
+	alt	keycode  41 = Meta_grave
+keycode  42 = Shift
+keycode  43 = dead_tilde       dead_circumflex   asciitilde  asciicircum
```
Comment (dmage) [inline data/keymaps/i386/hcesar/hcesar.map:80]: I don't see `~` on https://en.wikipedia.org/wiki/HCESAR. Why was it decided to put it on this key?

File: data/keymaps/i386/hcesar/hcesar.map [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+keycode  31 = q
+keycode  32 = t
+keycode  33 = d
+keycode  34 = i
+keycode  35 = n
+keycode  36 = u
+keycode  37 = l
+keycode  38 = m
+keycode  39 = x     
+	alt	keycode  39 = Meta_semicolon
```
Comment (dmage) [inline data/keymaps/i386/hcesar/hcesar.map:72]: Why 39 is `x`, but alt 39 is `M-;`? I don't see a key with `M-x`.

**Comment by mareuo** [comment]
Signed-off-by: Mario <random@developer.example.org>

File: data/keymaps/i386/hcesar/hcesar.map (hunk 4 above)
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:72]: my mistake

File: data/keymaps/i386/hcesar/hcesar.map (hunk 3 above)
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:80]: its there, same as pt-latin1.map.

File: data/keymaps/i386/hcesar/hcesar.map (hunk 2 above)
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:84]: its  +ccedilla         Ccedilla
on pt-latin1.map and pt-latin9.map.

File: data/keymaps/i386/hcesar/hcesar.map (hunk 1 above)
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:11]: no

File: data/keymaps/i386/hcesar/hcesar.map [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+keycode  20 = s 
+keycode  21 = a
+keycode  22 = r
+keycode  23 = o
+keycode  24 = p
+keycode  25 = z
+keycode  26 = plus             asterisk         dead_diaeresis
+	control keycode  26 = Escape
+	alt	keycode  26 = Meta_bracketleft
+keycode  27 = dead_acute       dead_grave       apostrophe       grave
```
[the remark this answers is not in the document]
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:56]: its the same as in pt-latin1.map. 
maybe it should be acute.

File: data/keymaps/i386/hcesar/hcesar.map (hunk 3 above)
Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:80]: for ã, õ, ñ.

Comment (mareuo) [reply data/keymaps/i386/hcesar/hcesar.map:78]: til

**Review by dmage — APPROVED** [review]

**Comment by legionus** [comment]
> ```
> Signed-off-by: Mario Oliveira <mauruon@proton.me>
> ```

Please add this line into the commit message. Other than that LGTM.

**Comment by legionus** [comment]
@mareuo You need to squash commits.

**Comment by legionus** [comment]
Applied with minor changes.

**Pull request closed without merging**